wgsl/index.bs

```diff
@@ -8973,20 +8985,55 @@ the implementation can issue a clear diagnostic.
   </xmp>
 </div>
 
-## Extensions List ## {#extension-list}
+## Software Extensions ## {#software-extensions-sec}
+
+A <dfn noexport>software extension</dfn> is an [=extension=] which is automatically available if the implementation supports it.
+The program does not have to explicitly request it.
+
+[=Software extensions=] embody functionality which could reasonably be supported on any WebGPU implementation.
+If the feature is not universally available, that it is because some WebGPU implementation has not yet implemented it.
+
+A WebGPU {{GPUAdapter}} signals support for a [=software extension=] by including the corresponding {{GPUFeatureName}}
```

Comment:
I'm not 100% clear on whether the intent is that GPUDevice must be created with an explicit request for the software-extension.

E.g. say there is a "tuples" extension, with corresponding GPUFeatureName "shader-tuples".
Does GPUDevice have to be created with "shader-tuples" in requiredFeatures for the functionality to be available?

Reply:
Question for @ben-clayton @kainino0x and others.

Reply:
This second commit is intended to fix #3875

Reply:
After internal discussion it isn't intended to be this way. I'll remove this bit.